### DISCUSSION TITLE
Rollback after custom task which is executed after deploy:symlink

### DISCUSSION THREAD

**Discussion by justm** [deployphp/deployer#2562] [Help needed]
Hello!
I have a custom task which is executed after deploy:symlink. 

```
task('deploy', [
     ...
    'build',
    'rsync',
    'deploy:shared',
    'deploy:writable',
    'deploy:clear_paths',

    'deploy:symlink',

    'my-custom-task'

    'deploy:unlock',
    'cleanup',
    'success'
]);
```

I would like to rollback to previous release if **my-custom-task** fails.
I use `fail('my-custom-task', 'rollback');` but that doesn't seem to work.

Custom task writes some file from remote server to "current" folder, that's why I am trying to run it after deploy:symlink.
Any help appreciated. Thanks!

**Comment by antonmedv** [comment]
Yes, current behaviour is to fail only on main task in this case is deploy. I think we can improve it.

**Reply by justm** [reply]
That would be great. What would be the planned milestone? Thanks

**Reply by antonmedv** [reply]
Can you implement it?

**Reply by tekiwebata** [reply]
Hello !
First of all, a big thank you for this very useful tool that is **Deployer**.
Any news on this feature ? I encountered the same behaviour : I want to do a specific task on another task on failure.
But it seems that the [line 178](https://github.com/deployphp/deployer/blob/ff69fd27e7e8fc808980923b812ad696ca3f9ff8/recipe/common.php#L178C1-L178C33) of the `common.php` recipe blocks the call of the fail function on task defined in the `deploy` task.
For exemple : the task `database:migrate` failed and I want to apply `database:migrate:rollback`. But, for now, the second task is not running because of `fail('deploy', 'deploy:failed');`.
Is it normal ? If yes, is there any plan to integer a correction for this "bug" ?
Thank you. Greetings.
